packages/appstore/src/components/demo-reset-balance/__tests__/demo-reset-balance.spec.tsx

```diff
@@ -0,0 +1,134 @@
+                </StoreProvider>
+            ),
+        });
+
+        expect(screen.getByRole('button', { name: /Reset balance/i })).toBeDisabled();
+    });
+
+    it('should call reset balance API when click on Reset balance', () => {
+        const mock = mockStore({
+            client: {
+                accounts: {
+                    VRW1002: {
+                        balance: 9880,
+                    },
+                },
+                loginid: 'VRW1002',
+            },
+        });
+        // @ts-expect-error need to come up with a way to mock the return type of useRequest
+        mockUseRequest.mockReturnValue({
+            mutate: jest.fn(),
+        });
+        const { mutate } = mockUseRequest('topup_virtual');
+        const setActiveTabIndex = jest.fn();
+
+        render(<DemoResetBalance setActiveTabIndex={setActiveTabIndex} />, {
+            wrapper: ({ children }) => (
+                <StoreProvider store={mock}>
+                    <APIProvider>{children}</APIProvider>
+                </StoreProvider>
+            ),
+        });
+
+        const reset_balance_button = screen.getByRole('button', { name: /Reset balance/i });
+        fireEvent.click(reset_balance_button);
+        expect(mutate).toBeCalledTimes(1);
+    });
+
+    it('should change tab when click on transfer funds button', () => {
+        const mock = mockStore({});
+        // @ts-expect-error need to come up with a way to mock the return type of useRequest
+        mockUseRequest.mockReturnValue({
+            isSuccess: true,
+        });
+        const setActiveTabIndex = jest.fn();
+
+        render(<DemoResetBalance setActiveTabIndex={setActiveTabIndex} />, {
+            wrapper: ({ children }) => (
+                <StoreProvider store={mock}>
+                    <APIProvider>{children}</APIProvider>
+                </StoreProvider>
+            ),
+        });
+
+        const transfer_funds_button = screen.getByRole('button', { name: /Transfer funds/i });
+
+        fireEvent.click(transfer_funds_button);
+        expect(setActiveTabIndex).toBeCalledTimes(1);
+    });
+
+    it('should show success message and transfer funds button if reset balance is reset successfully', () => {
+        const mock = mockStore({});
+        // @ts-expect-error need to come up with a way to mock the return type of useRequest
+        mockUseRequest.mockReturnValue({
+            isSuccess: true,
+        });
+        const setActiveTabIndex = jest.fn();
+
+        render(<DemoResetBalance setActiveTabIndex={setActiveTabIndex} />, {
+            wrapper: ({ children }) => (
+                <StoreProvider store={mock}>
+                    <APIProvider>{children}</APIProvider>
+                </StoreProvider>
+            ),
+        });
+
+        expect(screen.getByText('Your balance has been reset to 10,000.00 USD.')).toBeInTheDocument();
+        expect(screen.getByText('Transfer funds')).toBeInTheDocument();
```

Comment:
this should use getByRole as well since its a button

```suggestion
        expect(screen.getByRole('button', {name: 'Transfer funds'})).toBeInTheDocument();
```

Reply:
Updated.